eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/Recovery.scala

```diff
@@ -183,10 +182,7 @@ private class Acceptor(endpoint: ReplicationEndpoint) extends Actor {
     case re: ReplicationReadEnvelope if re.incompatibleWith(endpoint.applicationName, endpoint.applicationVersion) =>
       sender ! ReplicationReadFailure(IncompatibleApplicationVersionException(endpoint.id, endpoint.applicationVersion, re.targetApplicationVersion), re.payload.targetLogId)
     case ReplicationReadEnvelope(r, logName, _, _) =>
-      val r2 = endpoint.filters.get(r.targetLogId).orElse(endpoint.filters.get(logName)) match {
-        case Some(filter) => r.copy(filter = r.filter and filter)
-        case None         => r
-      }
+      val r2 = r.copy(filter = endpoint.endpointFilters.filterFor(r.targetLogId, logName) and r.filter)
```

Comment:
`and r.filter` can be removed once https://github.com/RBMHTechnology/eventuate/issues/297 is resolved, right?